source/common/mongo/proxy.h

```diff
@@ -73,6 +77,31 @@ class AccessLog {
 typedef std::shared_ptr<AccessLog> AccessLogSharedPtr;
 
 /**
+ * Mongo fault configuration.
+ */
+class FaultConfig {
```

Comment:
no need to have an interface.

Reply:
I would argue that this is more code. There is always a chance of a config being wrong but at least you'll be using the same code that production is using.

Reply:
it's more code when there is an interface and two implementations